gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/control/HeapMemoryMonitor.java

```diff
@@ -612,6 +631,46 @@ public void updateStateAndSendEvent(long bytesUsed) {
     }
   }
 
+  private boolean shouldDelayMemoryEvent(boolean delayMemoryEvent, MemoryState state, long bytesUsed) {
+    if (delayMemoryEvent && (state != null && state.isCritical())) {
+      // if still critical up but memory used has come down by 10% then reset the counters
+      long delta = bytesUsed - prevBytesUsed;
+      boolean isDownBy10percentOrMore = false;
+      if ( prevBytesUsed != 0 && delta < 0) {
+        long percent = (-delta/prevBytesUsed) * 100;
+        isDownBy10percentOrMore = percent > 10 ? true : false;
+      }
+      if (isDownBy10percentOrMore) {
```

Comment:
I think 10% decrease in just 1sec is unlikely. Perhaps it will be better to count prevBytesUsed since the last time countSinceUpsurge became > 0